public/js/sw-page-transition.js

```diff
@@ -76,11 +76,13 @@ const transition = (function() {
       this.log('fetchWithPageTransition: fetch ' + request.url);
       if (this._pagesBeingLoaded.includes(request.url)) {
         this.log('fetchWithPageTransition: returning cached page ' + request.url);
-        return caches.match(request).then(response => {
-          const index = this._pagesBeingLoaded.indexOf(request.url);
-          this._pagesBeingLoaded.splice(index, 1);
-          return response;
-        });
+        return this.openCache()
+          .then(cache => cache.match(request))
+          .then(response => {
+            const index = this._pagesBeingLoaded.indexOf(request.url);
+            this._pagesBeingLoaded.splice(index, 1);
```

Comment:
This is to support Gulliver's network first strategy -> only return the cached version if a loading page has been requested before. This is not a perfect solution as it can't handle (theoretical) parallel requests to the same page.